src/domain/repositories/DataSetsRepository.ts

```diff
@@ -1,7 +1,8 @@
 import { DataSet } from "domain/entities/DataSet";
-import { Id } from "types/d2-api";
+import { Id, MetadataResponse } from "types/d2-api";
 
 export interface DataSetsRepository {
     get(ids: Id[]): Promise<Record<Id, DataSet>>;
+    post(data: object, options: object): Promise<MetadataResponse>;
```

Comment:
`object` is too generic (only to use when you don't interpret the entity, but here we have a `DataSet`). 

The same applies to `options`, we should specific supported options. Typically, we don't expose all the d2-api options, only the ones we need.

Reply:
Switched to: `post(data: DataSetMetadata): Promise<OUCopyResult>;`.